lib/submit-analytics.js

```diff
@@ -33,50 +103,173 @@ function sha256(data) {
     return hash.digest('hex');
 }
 
-if (isAnalyticsDisabled()) {
-    module.exports = function(){};
-} else {
+/**
+ * Send collected analytics data to Realm's servers over HTTPS
+ * @param  {[WebhookProdDetails|WebhookStageDetails|MixpanelDetails]} requestFactory Submission URL generator
+ * @param  {Object} payload analytics info
+ */
+async function dispatchAnalytics(requestFactory, payload) {
+    const https = require('https');
+
+    return new Promise((resolve, reject) => {
+        const webhookRequest = new requestFactory().buildRequest(payload);
+        var req = https.get(webhookRequest, function (result) {
+            result.on('data', (data) => {
+                // we don't log the data that is returned -- we're just interested in the header
+            });
+        });
+
+        req.on('response', response => {
+            response.on('end', () => {
+                resolve(
+                    {
+                        statusCode:  response.statusCode,
+                        statusMessage: response.statusMessage,
+                        errorMessage: '',
+                    });
+                return;
+            });
+
+            response.on('error', error => {
+                reject(
+                    {
+                        statusCode:  response.statusCode,
+                        statusMessage: response.statusMessage,
+                        errorMessage: error,
+                    });
+                return;
+            });
+        });
+    });
+}
+
+/**
+ * Collect analytics data from the runtime system
+ * @param  {Object} context contents of the app's package.json file
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns {Object} Analytics payload
+ */
+function fetchPlatformData(context, eventName) {
```

Comment:
I suggest making this an async function and `await` the promise returned by calling `machineId()`

```suggestion
async function fetchPlatformData(context, eventName) {
```

Reply:
Fixed in #3524 .